### ISSUE TITLE
Make States optional

### ISSUE THREAD

**Issue by robinpokorny** [agudulin/dumb-bem#6]
The state class names like `is-hidden` are not part of the BEM. Yet, they are applied automatically when a prop with one of these (rather common) names is included.

I propose it is removed from default behaviour. When someone wants to use that they can pass it as an extra maker. It could look like this:

``` js
import dumbBem from 'dumb-bem'
import makeStates from 'dumb-bem/makers/makeStates'

const dumbHeader = dumbBem('header', [makeStates])
```

When this would be desired behavior in a codebase such middle module could be written:

``` js
import dumbBem from 'dumb-bem'
import makeStates from 'dumb-bem/makers/makeStates'

export default (block) => dumbBem(block, [makeStates])
```

The last line could read the following to enable extra makers:

``` js
export default (block, extraMakers = [] ) =>
  dumbBem(block, [makeStates].concat(extraMakers))
```

**Comment by robinpokorny** [comment]
I was thinking, when something does not depend on the `block` then what is the reason to use dumb-bem for it. Then we can pass it directly to `tx`. Like so

``` js
const dumbSmacss ({ className = [], ...props }) => {
  const className = ['active', 'disabled', 'hidden', 'loading']
    .filter((s) => props[s])
    .map((s) => `is-${s}`)
    .concat(className)
    .join(' ')

  return { ...props, className }
}

const dumbHeader = dumbBem('header')
const HeaderTitle = tx([dumbHeader, { element: 'title' }, dumbSmacss])('h1')
```

Unless we would like to handle all class name generation in this package, then we might discuss about renaming it.

What do you say, @agudulin, @paranoida ?

(BTW Originally `dumbSmacss` was `statesToClass`; the former just sounds funnier.)

**Comment by rafalbromirski** [comment]
Makes sense. many people are using `block--disabled` or just `block-disabled`.

I believe we should extract all states and create `dumb` helper inside our utils. Other solution would be to have:
- dumb-bem-core
- dumb-bem-smacss

;)

**Comment by agudulin** [comment]
Totally agree.

I like how babel is organized: it has a core, and a lot of packages: https://github.com/babel/babel/tree/master/packages

Looks like @paranoida's idea for naming.

**Comment by agudulin** [comment]
Implemented in v3.0.0
